File: field_uniform.py
import numpy as np

# Constants
B0 = 1.0
m = 1.0
qe = 1.0
c = 1.0

# Functions
def Bmod(x):
    return B0

def dBmod(x):
    return 0.0

def A(x):
    return B0*np.array([0.0, x[0], 0.0])

def dA(x):
    dA = np.zeros((3, 3))
    dA[0, 1] = B0
    return dA

def B(x):
    return np.array([0.0, 0.0, B0])

def dB(x):
    return np.zeros((3,3))

def e1(x):
    return np.array([0.0, 1.0, 0.0])

def e2(x):
    return np.array([1.0, 0.0, 0.0])

def de1(x):
    return np.zeros((3,3))

def de2(x):
    return np.zeros((3,3))

File: field_uniform.jl
const B0 = 1.0

function Bmod(x)
    return B0
end

function dBmod(x)
    return [0.0, 0.0, 0.0]
end

function A(x)
    return B0 .* [0.0, x[1], 0.0]
end

function dA(x)
    dA = zeros(3, 3)
    dA[1, 2] = B0
    return dA
end

function B(x)
    return [0.0, 0.0, B0]
end

function dB(x)
    return zeros(3, 3)
end

function e1(x)
    return [0.0, 1.0, 0.0]
end

function e2(x)
    return [1.0, 0.0, 0.0]
end

function de1(x)
    return zeros(3, 3)
end

function de2(x)
    return zeros(3, 3)
end
